doc/security/sensor-threat.rst

```diff
@@ -0,0 +1,327 @@
+.. _sensor-threat:
+
+Sensor Device Threat Model
+##########################
+
+This document describes a threat model for an IoT sensor device. This
+device contains a sensor of some type (for example temperature, or a
+pressure in a pipe), which sends this data to an SoC running a
+microcontroller. This microcontroller connects to a cloud service, and
+relays this sensor data to this service. The cloud service is also able
+to send configuration data to the device, as well as software update
+images. A general diagram can be seen in Figure 1:
+
+.. figure:: media/sensor-model.svg
+
+   Figure 1. Sensor General Diagram
+
+In this sensor device, the sensor connects with the SoC via an SPI bus,
+and the SoC has a network interface that it uses to communicate with the
+cloud service. The particulars of these interfaces can impact the threat
+model in unexpected ways, and variants on this will need to be
+considered (for example, using a separate network interface SoC
+connected via some type of bus).
+
+This model also focuses on communicating via the MQTT-over-TLS protocol,
+as this seems to be in wide use [1]_.
+
+Assets
+======
+
+One aspect of the threat model to consider are assets involved in the
+operation of the device. The following list enumerates the assets
+included in this model:
+
+1. **The bootloader**. This is a small code/data image contained in
+   on-device flash that is the first code to run. In order to establish
+   a root of trust, this image must be immutable. This model assumes
+   that the SoC provides a mechanism to protect a region of the flash
+   from future writes, and that this will be done after this image is
+   programmed into the device, early in production.
+
+2. **The application firmware image**. This asset consists of the
+   remainder of the firmware run by the microcontroller. The distinction
+   is made because this part of the image will need to be updated
+   periodically as security vulnerabilities are discovered. Requirements
+   for updates to this image are:
+
+   a. The image shall only be replaced with an authorized image.
+
+   b. When an authorized replacement image is available, the update
+      shall be done in a timely manner.
+
+   c. The image update shall be seen as atomic, meaning that when the
```

Comment:
Clearly, the application image may consist of multiple binary imaged merged together to form a single firmware. That said, it is possible that firmware upgrade concerns only _parts_ of the application image, so it is good to incorporate this in the text, somehow.

E.g. something like
 "even if a subset of the FW components are to be updated, this is a new _logical_ image and shall be updated in its entirety (as a single _logical_ image)"

Reply:
It could be possible to upgrade the image partially, although this gets rather complicated. This is an open issue in MCUboot on how this can be done when we have a secure and non-secure image. In generally, they either need to both be upgraded, or both not upgraded.

This doesn't preclude optimizations that only download changes to the image.